test/unit/uk/gov/hmrc/tariffclassificationfrontend/service/CasesService_SuppressCaseSpec.scala

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tariffclassificationfrontend.service
+
+import org.mockito.ArgumentMatchers.{refEq, _}
+import org.mockito.BDDMockito._
+import org.mockito.Mockito.{never, reset, verify, verifyZeroInteractions}
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.mockito.MockitoSugar
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.test.UnitSpec
+import uk.gov.hmrc.tariffclassificationfrontend.audit.AuditService
+import uk.gov.hmrc.tariffclassificationfrontend.config.AppConfig
+import uk.gov.hmrc.tariffclassificationfrontend.connector.BindingTariffClassificationConnector
+import uk.gov.hmrc.tariffclassificationfrontend.models._
+import uk.gov.hmrc.tariffclassificationfrontend.models.request.NewEventRequest
+import uk.gov.tariffclassificationfrontend.utils.Cases
+
+import scala.concurrent.Future.{failed, successful}
+
+class CasesService_SuppressCaseSpec extends UnitSpec with MockitoSugar with BeforeAndAfterEach with ConnectorCaptor {
+
+  private implicit val hc: HeaderCarrier = HeaderCarrier()
+
+  private val manyCases = mock[Seq[Case]]
+  private val oneCase = mock[Option[Case]]
+  private val connector = mock[BindingTariffClassificationConnector]
+  private val emailService = mock[EmailService]
+  private val fileStoreService = mock[FileStoreService]
+  private val audit = mock[AuditService]
+  private val config = mock[AppConfig]
+  private val aCase = Cases.btiCaseExample
+
+  private val service = new CasesService(config, audit, emailService, fileStoreService, connector)
+
+  override protected def afterEach(): Unit = {
+    super.afterEach()
+    reset(connector, audit, oneCase, manyCases, config)
+  }
+
+  "Suppress a Case" should {
```

Comment:
it is fine to use `any[HeaderCarrier]`, but for the other arguments, I would prefer using `refEq`, so that we make our tests stricter

Reply:
Have used refEq for cases but it's harder for NewEventRequest because that case class contains a default of `Instant.now()`  for the 'timestamp' field

Reply:
@gokyo He's using argument captors below to assert on the content of the Event which I think is the best approach

Reply:
If we overuse `refEq` we end up getting `NullPointer`s all over the place making tests hard to debug. Only my personal opinion but I'd prefer for non trivial stuff like this we use Argument Captors like Nick has below

Reply:
I've based my test on an existing test. If we have a wider problem across existing tests maybe we should capture the problem on a tech debt card.